plexpy/activity_handler.py

```diff
@@ -360,114 +360,38 @@ def get_metadata(self):
-
-                        # Schedule a callback to clear the recently added queue
-                        schedule_callback('rating_key-{}'.format(parent_rating_key),
-                                          func=clear_recently_added_queue,
-                                          args=[parent_rating_key, parent_title],
-                                          seconds=plexpy.CONFIG.NOTIFY_RECENTLY_ADDED_DELAY)
-
-                else:
-                    queue_set = RECENTLY_ADDED_QUEUE.get(rating_key, set())
-                    RECENTLY_ADDED_QUEUE[rating_key] = queue_set
-
-                    logger.debug(u"Tautulli TimelineHandler :: Library item '%s' (%s) added to recently added queue."
-                                 % (title, str(rating_key)))
-
-                    # Schedule a callback to clear the recently added queue
-                    schedule_callback('rating_key-{}'.format(rating_key),
-                                      func=clear_recently_added_queue,
-                                      args=[rating_key, title],
-                                      seconds=plexpy.CONFIG.NOTIFY_RECENTLY_ADDED_DELAY)
-
-            # A movie, show, or artist is done processing
-            elif media_type in ('movie', 'show', 'artist') and section_id > 0 and \
-                state_type == 5 and metadata_state is None and queue_size is None and \
-                rating_key in RECENTLY_ADDED_QUEUE:
-
-                logger.debug(u"Tautulli TimelineHandler :: Library item '%s' (%s) done processing metadata."
-                             % (title, str(rating_key)))
+            state_type = self.timeline.get('state')
+            metadata_state = self.timeline.get('metadataState')
 
             # An item was deleted, make sure it is removed from the queue
-            elif state_type == 9 and metadata_state == 'deleted':
+            if state_type == 9 and metadata_state == 'deleted':
                 if rating_key in RECENTLY_ADDED_QUEUE and not RECENTLY_ADDED_QUEUE[rating_key]:
                     logger.debug(u"Tautulli TimelineHandler :: Library item %s removed from recently added queue."
-                                 % str(rating_key))
-                    del_keys(rating_key)
-
-                    # Remove the callback if the item is removed
-                    schedule_callback('rating_key-{}'.format(rating_key), remove_job=True)
-
+                         % str(rating_key))
+                del_keys(rating_key)
+                
+                # Remove the callback if the item is removed
+                schedule_callback('rating_key-{}'.format(rating_key), remove_job=True)
+            
+            else:
+                logger.debug(u"Tautulli TimelineHandler :: Library item %s scheduling process with delay."
+                             % str(rating_key))
+            
+                schedule_callback('rating_key-{}'.format(rating_key),
```

Comment:
You have a callback for every single item. How do you reset the callback timer when you add another episode to the same season in order to group them together in one notification?

i.e. Your notification delay is 10 minutes. You add one episode of a TV show and the callback is scheduled for 10 minutes. You add another episode 9 minutes later. The callback needs to reset to 10 minutes again.